File: java/src/televentas/EmpresaTransporte.java
package televentas;

public class EmpresaTransporte {

    private final int idEmpresa;
    private final String nombre;
    private final String cobertura;

    public EmpresaTransporte(int idEmpresa, String nombre, String cobertura) {
        this.idEmpresa = idEmpresa;
        this.nombre = nombre;
        this.cobertura = cobertura;
    }

    public int getIdEmpresa() {
        return idEmpresa;
    }

    public String getNombre() {
        return nombre;
    }

    public String getCobertura() {
        return cobertura;
    }

    public double calcularTarifa() {
        return 15000.0;
    }

    public void asignarEnvio(Envio envio) {
        envio.programar();
    }
}

File: javascript/src/televentas/empresaTransporte.js
export class EmpresaTransporte {
  constructor(idEmpresa, nombre, cobertura) {
    this.idEmpresa = idEmpresa;
    this.nombre = nombre;
    this.cobertura = cobertura;
  }

  getIdEmpresa() {
    return this.idEmpresa;
  }

  getNombre() {
    return this.nombre;
  }

  getCobertura() {
    return this.cobertura;
  }

  calcularTarifa() {
    return 15000.0;
  }

  /** @param {import('./envio.js').Envio} envio */
  asignarEnvio(envio) {
    envio.programar();
  }
}
